docs/map-making/how-to-upgrade-legacy-map-features.md

```diff
@@ -35,70 +19,36 @@ All occurrences of the code before the `>>` must be substituted with the code af
 - `="conditionType" count="XOR"` >> `="conditionType" count="1"`
 - `="conditionType" count="-reset-XOR"` >> `="conditionType" count="-reset-1"`
 - `="isTwoHit" count="true"` >> `="hitPoints" count="2"`
-- `="isTwoHit" value="false"` >> `="hitPoints" value="1"`
-- `="isTwoHit" count="-reset-true"` >> `="hitPoints" count="-reset-2"`
 - `="isTwoHit" count="false"` >> `="hitPoints" count="1"`
+- `="isTwoHit" value="true"` >> `="hitPoints" value="2"`
+- `="isTwoHit" value="false"` >> `="hitPoints" count="1"`
+- `="isTwoHit" count="-reset-true"` >> `="hitPoints" count="-reset-2"`
 - `="isTwoHit" count="-reset-false"` >> `="hitPoints" count="-reset-1"`
 - `="occupiedTerrOf"` >> `="originalOwner"`
 - `="victoryCity" count="true"` >> `="victoryCity" count="1"`
-- `="victoryCity" count="-reset-true"` >> `="victoryCity" count="-reset-1"`
 - `="victoryCity" count="false"` >> `="victoryCity" count="0"`
+- `="victoryCity" value="true"` >> `="victoryCity" count="1"`
+- `="victoryCity" value="false"` >> `="victoryCity" count="0"`
+- `="victoryCity" count="-reset-true"` >> `="victoryCity" count="-reset-1"`
 - `="victoryCity" count="-reset-false"` >> `="victoryCity" count="-reset-0"`
-
-
-### **Procedural alternatives:**
-
-For a more defined change, instead of the aforementioned:
-
-- `attatchment` >> `attachment`
-
-you can change:
-
-- `attatchmentList` >> `attachmentList`
-- `<attatchment` >> `<attachment`
-- `attatchment>` >> `attachment>`
-- `="games.strategy.triplea.attatchments` >> `="games.strategy.triplea.attachments`
-
-- `="techAttatchment` >> `="techAttachment`
-- `="unitAttatchment` >> `="unitAttachment`
-- `="territoryAttatchment` >> `="territoryAttachment`
-- `="canalAttatchment` >> `="canalAttachment`
-- `="rulesAttatchment` >> `="rulesAttachment`
-- `="playerAttatchment` >> `="playerAttachment`
+- `<option name="isParatroop"` >> `<option name="isAirTransportable"`
+- `<option name="isMechanized"` >> `<optiona name="isLandTransportable"`
 
 
 ### XML Deletions
 
-_(these items were already deprecated and did nothing)_
-
-- `<option name="isParatroop"*/>`
-- `<option name="isMechanized"*/>`
-- `<option name="takeUnitControl"*/>`
-- `<option name="giveUnitControl" value="true"/>`
-- `<option name="giveUnitControl" value="false"/>`
+Delete these options entirely:
 
-## map.properties Changes
+- `<option name="takeUnitControl"`
+- `<option name="giveUnitControl"`
```

Comment:
As I said in a section deleted at this update "any option named "giveUnitControl" must be deleted only if its value is "true" or "false".

If this is not true, this needs to be indagated, as I believe "giveUnitControl" is currently documented as an effective and non-deprecated option. If so, telling to delete any occurrence of such option would be a very bad and damaging information.

Reply:
Why was this removal mentioned in the original notes?  I'm not aware we removed the 'true' 'false' to do nothing in 1.8 or 1.9

Reply:
I think it's better we just remove it for now. Chances are good the map will crash with incorrect values which would mean it would get noticed. 

Updated: 131e6407a

Reply:
Absolutely not. As fas as I know, "giveUnitControl" is a currently valid option that should not necessarily be deleted.

What my notes were saying is that "giveUnitControl" was doing nothing if set at value "true" or "false" and that, after the removals, should be deleted if set at those values (only).

If my notes were wrong, a developer would need to prove it by looking at the code or testing. However, I can reference the official game information, that currently have "giveUnitControl" as a non-deprecated option:
https://github.com/triplea-maps/the_pact_of_steel/blob/master/map/games/pact_of_steel_2.xml#L4021

Therefore, unless proved wrong, my information that "takeUnitControl" should be always deleted (as an option), while "giveUnitControl" should be deleted (as an option) only if set "true" of "false" should be correct and valuable information. Removing all reference to such options would be, in this case, a major step backward in making this document less incomplete.

Reply:
> What my notes were saying is that "giveUnitControl" was doing nothing if set at value "true" or "false" and that, after the removals, should be deleted if set at those values (only).

Why would it have been set to those values? Was it doing something previously? If so, I'm not aware of that changing in 1.9, was it change in 1.9? 

Are you saying "always" deleted assuming someone upgraded to 1.8 correctly? Personally, I am _NOT_ assuming maps were always correctly upgraded to a 1.8 and just because you see a 'does nothing' property in 1.8 that it should be removed rather than replaced by it's correct replacement.

Reply:
Right or wrong (and, to the extent of my knowlege, it has always been right), what my documentation was telling is to:
- Always delete the "takeUnitControl" option.
- Delete the "giveUnitControl" option only if its value is "true" or "false".

I think this is clear enough.

This is based on the information (which now it is an assumption) that such items to delete were doing nothing, of course.

Reply:
Indeed, dangerous assumption. You're assuming that someone:
1. was upgrading their map
2. saw a deprecated property
3. knew it was supposed to be updated
4. Added the new property but did not remove the old

IMO it's MUCH more likely that a map was simply just *NEVER* updated and so the property still remains.

When we consider the risk of the two scenarios:
- in the first you have unclean XML and possibly conflicting values. If a map maker follows this guide then they'll see they have duplicate values (hopefully, but the proximity will be close to each other) and will resolve).
- in the second you have a map that is just not working correctly.

I'll emphasize this is a change log document now to describe what map makers should do *now* given whatever map they are seeing. This goes back to my point about the lack of utility to upgrade to only 1.9.x and also how that does not serve an ongoing purpose of where to document changes (which then if this is a historical document, we then have duplication, legacy issues and just essentially more documentation laying around that makes it all harder to keep up to date).

Reply:
>  Otherwise, for example in this case, if I'm correct, you would need to have one upgrade information set for when the option was doing one thing and another one for when the option was doing nothing.

I don't think so. If the new properties would lead to duplication, we can leave it to a map maker to see that duplication. Even if they do not spot it, the duplication causes no harm. If the values were conflicting, then the map had issues to begin with.

> Whether or not the map was correctly coded in 1.8.0.9 I believe it is beside the point.

That actually is very much to the point. If we say "delete" this, and the map then changes in a way that is hard to notice, that's bad. Compare that to the risk of something not being upgraded more properly (likely), or someone not realizing that they are looking at a 1.7 map thinking it is a 1.8 map, then suddenly our directions steer them wrong.

Reply:
I'm not going to add these back in:

```

    Always delete the "takeUnitControl" option.
    Delete the "giveUnitControl" option only if its value is "true" or "false".
```

We need to know what the options are supposed to be changed into and let a map maker take care of potential duplication (which is relatively harmless if there is a duplicated value).

Reply:
Well, the original documentation was meant for upgrading maps from 1.8.0.9, while keeping exactly the same behaviour they had, when played with that version of TripleA. If now you are setting different standards, that is a change of direction. Here I'm just highlighting risks. However, if it can be confirmed that, at some time in TripleA history, these options were changed to become ineffective (and this was still the case when 1.8.0.9 came out), I think it would be the only consistent thing to do to turn them ineffective still. It is the same matter as if an option's behaviour was changed in the past. Maybe there is a map still coded on the assumption that the old behaviour applies, but you cannot go after such things, reliably.

Reply:
Please see my other comments, it's more risky to assume that someone did a bad upgrade to 1.8.0.9 and left in needless properties after replacing rather than the risk of they simply never upgraded at all.

If we need to assume a perfect upgrade to 1.8.0.9, then we need to have that documentation, which we don't. Otherwise how can you ever be confident, even if someone claims that a given map was upgraded, that it was done in a correct manner?

Reply:
> Well, the original documentation was meant for upgrading maps from 1.8.0.9

I am changing the scope, in sum for three reasons:
- that is a risky assumption (see previous comment)
- not useful (there is no point to upgrade to just 1.9.0.0.3xxx and there is no guarantee that a map is actually perfectly on 1.8.0.9)
- fails to serve an ongoing need